api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -445,6 +448,45 @@ private void verifyInvitationKey(String invitationKey){
     }
   }
 
+  @Override
+  public ResponseEntity<Void> updateContactEmail(UpdateContactEmailRequest updateContactEmailRequest) {
+    verifyNoLogins(updateContactEmailRequest.getUsername());
+    //resend Welcome Email to new contact email... Add once Mandrill API complete
+    updateUser(updateContactEmailRequest);
+    return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+  }
+
+  private void verifyNoLogins(String username){
+    User user = userDao.findUserByEmail(username);
+    if (user.getFirstSignInTime() != null) {
+      throw new BadRequestException(
+        "This account has already been in use, if you would like to update your contact email please login and update via the Profile page."
+      );
+    }
+  }
+
+  private void updateUser(UpdateContactEmailRequest updateRequest) {
+    com.google.api.services.admin.directory.model.User googleUser =
+      directoryService.getUser(updateRequest.getUsername());
+    String contactEmail = updateRequest.getContactEmail();
+//    try {
```

Comment:
I commented this out while I was working on the other issues. But we need this, or something like it to validate updated email addresses...